A workaround for Exchange 2010 IMAP protocol, tracked in
http://bugs.mutt.org/3459, already merged in the ubuntu version of mutt

--- a/imap/message.c
+++ b/imap/message.c
@@ -242,6 +242,15 @@
       char *cmd;
 
       fetchlast = msgend + 1;
+      /* Microsoft Exchange 2010 violates the IMAP protocol and
+       * starts omitting messages if one FETCHes more than 2047 (or
+       * or somewhere around that number. We therefore split the
+       * FETCH into chunks of 2000 messages each */
+      if (fetchlast - msgno - 1 > 2000)
+      {
+        fetchlast = msgno+1 + 2000;
+      }
+
       safe_asprintf (&cmd, "FETCH %d:%d (UID FLAGS INTERNALDATE RFC822.SIZE %s)",
                      msgno + 1, fetchlast, hdrreq);
       imap_cmd_start (idata, cmd);
